File: Classe extendida Fila e Pilha/C++/pilha.cpp
#include <iostream>
#include <stack>
#include "lista.h"
#include "pilha.h"

int Pilha::first(){
    if(this->pilha.empty()){
        std::cout << "Pilha vazia" << std::endl;
        return -1;
    } else return this->pilha.top();
}

void Pilha::insere(int value){
    this->pilha.push(value);
    this->incElem();
}

void Pilha::retira(){
    if(this->pilha.empty()){
        std::cout << "Pilha vazia" << std::endl;
        return;
    } else this->pilha.pop(); this->decElem();
}

File: Classe extendida Fila e Pilha/Java/Pilha.java
import java.util.Stack;

public class Pilha extends Lista {
    private Stack<Integer> pilha;

    public Pilha(){
        super();
        this.pilha = new Stack<>();
    }

    public int first(){
        if (this.pilha.empty()){
            System.out.println("pilha vazia\n");
            return -1;
        } else return this.pilha.peek();
    }

    public void insere(int value){
        this.pilha.push(value);
        this.incElem();
    }

    public int retira(){
        this.decElem();
        return this.pilha.pop();
    }
}